lib/util/sdp/index.js

```diff
@@ -251,6 +289,84 @@ function setSimulcast(sdp, sdpFormat, trackIdsToAttributes) {
   })).concat('').join('\r\n');
 }
 
+/**
+ * Filter local codecs based on the remote unified plan SDP.
+ * @param {string} localSdp
+ * @param {string} remoteSdp
+ * @returns {string} - Updated local SDP
+ */
+function unifiedPlanFilterLocalCodecs(localSdp, remoteSdp) {
+  const localMediaSections = getMediaSections(localSdp);
+  const localSession = localSdp.split('\r\nm=')[0];
+  const remoteMidsToMediaSections = createMidToMediaSectionMap(remoteSdp);
+  return [localSession].concat(localMediaSections.map(localSection => {
+    const localMid = getMidForMediaSection(localSection);
+    const remoteSection = localMid && remoteMidsToMediaSections.get(localMid);
+    // Do nothing if either the local m= section does not have an equivalent
+    // remote m= section or if it is neither an audio nor a video m= section.
+    if (!(remoteSection && /^m=(audio|video)/.test(localSection))) {
```

Comment:
you can optimize a bit by returning early if `localSection` was not video or audio.

```
  // do nothing if neither an audio nor a video m= section.
  if (/^m=(audio|video)/.test(localSection) === false) {
     return localSection;
  }

   const localMid = getMidForMediaSection(localSection);
   const remoteSection = localMid && remoteMidsToMediaSections.get(localMid);
   if (!remoteSection) {
    // the local m= section does not have an equivalent
    // remote m= section
    return localSection;
   }

```